content/blog/2019/egress-traffic-control-in-istio-part-2/index.md

```diff
@@ -0,0 +1,101 @@
+provided outside of Istio by the cluster/cloud provider).
+The L3 firewall can have a trivial configuration that would allow incoming traffic into Istio ingress gateway pods only and
+outgoing traffic from Istio egress gateway pods only. Note that the Istio proxy of the egress gateway performs
+policy enforcement and reporting in the same way as the sidecar proxies in the application pods.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewall" >}}
+
+Now let's examine possible attacks and let me show you how the secure control of egress traffic in Istio prevents them.
+
+## Possible attacks and their prevention
+
+Consider the following security policy with regard to egress traffic:
+
+1. Application A is allowed to access `*.ibm.com` (all the external services with URL matching `*.ibm.com`,
+  e.g. `www.ibm.com`)
+1. Application B is allowed to access `mongo1.composedb.com`
+1. All egress traffic must be monitored
+
+Now consider a scenario in which one of application A's pods is compromised. Suppose the attackers have the
+following goals:
+
+1. Application A will try to access `*.ibm.com` unmonitored
+1. Application A will try to access `mongo1.composedb.com`
+
+Note that application A is allowed to access `*.ibm.com`, so the attacker is able to access it. There is no way
+to prevent such access since there is no way to distinguish, at least initially, between the original and the
+compromised versions of application A. However, you want to monitor any access to external services to be able to
+detect suspicious traffic, for example by applying anomaly detection tools on logs of the egress traffic.
+The attackers, on the contrary, want to access external services unmonitored, so the attack will not be detected.
+The second goal of the attackers is to access `mongo1.composedb.com`, which is forbidden for application A. Istio
+must enforce the policy that forbids access of application A to `mongo1.composedb.com` and must prevent the attack.
+
+Now let's see which attacks the attackers will try to perform to achieve their goals and how Istio secure egress traffic
+control will prevent each kind of attack. The attackers may try to:
+
+1. **Bypass** the container's sidecar proxy and access external services directly. This attack is prevented by a
+   Kubernetes Network Policy or by an L3 firewall that allow egress traffic to exit the mesh only from the egress
+   gateway.
+1. **Compromise** the egress gateway and force it to send fake information to the monitoring system or to disable
+   enforcement of the security policies.
+   This attack is prevented by applying the special security measures to the egress gateway pods.
+1. Since the previous attacks are prevented, the attackers have no other option but to direct the traffic through the
```

Comment:
@rcaballeromx _the attackers hope it will go undetected_ - the attackers cannot be so naive :) We are talking about the attackers who learned how Istio works and are trying to break various Istio mechanisms.